### ISSUE TITLE
InlineKeyboardButton only works with the :url key

### ISSUE THREAD

**Issue by dotboris** [zhyu/nadia#44]
I'm trying to send a message with some inline keyboard buttons.

According to Telegram's [documentation](https://core.telegram.org/bots/api#inlinekeyboardbutton). When specifying a `InlineKeyboardButton` I should pass only one of the optional keys.

For some reason, I can only get the `url` key to work.

The following code works:

```elixir
button = %Nadia.Model.InlineKeyboardButton{text: "test", url: "https://google.com"}
markup = %Nadia.Model.InlineKeyboardMarkup{inline_keyboard: [[button]]}
Nadia.send_message(chat_id, "Hello?", reply_markup: markup)
```

The chat message will show with a working link button.

If I use something like `callback_data` instead of `url` I get an error complaining about the `url` key not being a string.

```elixir
button = %Nadia.Model.InlineKeyboardButton{text: "test", callback_data: "something"}
markup = %Nadia.Model.InlineKeyboardMarkup{inline_keyboard: [[button]]}
Nadia.send_message(chat_id, "Hello?", reply_markup: markup)
```

The error:

```elixir
{:error,
 %Nadia.Model.Error{reason: 
   "Bad Request: can't parse inline keyboard button: Field \"url\" must be of type String"}}
```

I'm not sure if this is a problem with nadia or if this is an issue with the Telegram bot api.

**Comment by dotboris** [comment]
I've done some digging and I think that the JSON serialization might be to blame. If you serialize the `InlineKeyboardButton` struct with `Poison.encode!` you get a json blob with all non-specified keys set to null.

For example:

```elixir
%Nadia.Model.InlineKeyboardButton{text: "foo", callback_data: "bar"}
|> Poison.encode!
|> IO.puts
#=> {"url":null,"text":"foo","switch_inline_query":null,"callback_data":"bar"}
```

I'm guessing that Telegram expects one of the keys to be specified at all instead of being null.

**Comment by dotboris** [comment]
I've done some tests and I think that my hypothesis is correct.

I implemented `Poison.Encoder` for `InlineKeyboardButton` and removed all keys with nil values before encoding. After doing so, my previously broken code started working.

I have implemented this in my own fork. Right now it's pretty dirty but it does work. This seems like something that would affect other types in the api. Ideally, I should also implement this for those as well.

Everything is in the following branch <https://github.com/dotboris/nadia/tree/remove-nil-values-in-markup>
